storage/integration_test.go

```diff
@@ -3888,6 +3888,261 @@ func TestIntegration_LockBucket_MetagenerationRequired(t *testing.T) {
 	})
 }
 
+func TestIntegration_BucketObjectRetention(t *testing.T) {
+	ctx := skipJSONReads(skipGRPC("not yet available in gRPC - b/308194853"), "no reads in test")
+	multiTransportTest(ctx, t, func(t *testing.T, ctx context.Context, _ string, prefix string, client *Client) {
+		setTrue, setFalse := true, false
+
+		for _, test := range []struct {
+			desc                string
+			objectRetentionMode string
+			enable              *bool
+			wantRetentionMode   string
+		}{
+			{
+				desc:              "ObjectRetentionMode is not enabled by default",
+				wantRetentionMode: "",
+			},
+			{
+				desc:                "Setting just ObjectRetentionMode should not work",
+				objectRetentionMode: "Enabled",
+				wantRetentionMode:   "",
+			},
+			{
+				desc:                "ObjectRetentionMode is ignored",
```

Comment:
I'm really confused by this case (and somewhat by the case above as well) and it seems very counterintuitive behavior. We should discuss with the product team.